modules/upload/upload.service.js

```diff
@@ -0,0 +1,119 @@
+
+class UploadService {
+    async uploadFiles(files) {
+        return files.map(async file => {
+            const { createReadStream, filename } = await file;
+
+            const inputStream = createReadStream()
+            let fileBuffer;
+            const id = uniqid()
+
+            const inputBuffer = await new Promise((resolve, reject) => {
+                let chunks = [];
+                inputStream.once('error', (err) => {
+                    return reject(err);
+                });
+
+                inputStream.once('end', () => {
+                    fileBuffer = Buffer.concat(chunks);
+                    return resolve(fileBuffer);
+                });
+
+                inputStream.on('data', (chunk) => {
+                    chunks.push(chunk);
+                });
+            })
+
+            Jimp.read(inputBuffer).then(image => {
+                image.resize(1920, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
+                    const
+                        blobName = `large_${id}_${filename}`,
+                        stream = getStream(buffer),
+                        streamLength = buffer.length
+
+
+                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
+                        if (err) {
+                            throw new Error(err)
+                        }
+                    })
+                })
+                image.resize(1080, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
+                    const
+                        blobName = `medium_${id}_${filename}`,
+                        stream = getStream(buffer),
+                        streamLength = buffer.length
+
+                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
+                        if (err) {
+                            throw new Error(err)
+                        }
+                    })
+                })
+
+                image.resize(768, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
+                    const
+                        blobName = `small_${id}_${filename}`,
+                        stream = getStream(buffer),
+                        streamLength = buffer.length
+
+                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
+                        if (err) {
+                            throw new Error(err)
+                        }
+                    })
+                })
+
+                image.resize(128, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
+                    const
+                        blobName = `thumbnail_${id}_${filename}`,
+                        stream = getStream(buffer),
+                        streamLength = buffer.length
+
+                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
```

Comment:
Please, convert `createBlockBlobFromStream` to `Promise` using `promisify` from the default node package `util`

Reply:
fixed